### ISSUE TITLE
Docs missing encoding example

### ISSUE THREAD

**Issue by jsteer** [HCA-Healthcare/elixir-hl7#3]
If the library is capable of doing encoding?, an example and/or test for encoding  a message starting from a filled in struct (like MSH) would be very helpful.

**Comment by DarkMarmot** [comment]
It is -- though it's definitely not straight-forward right now. We'd actually been discussing breaking this library up into separate libraries: one for parsing/mutating/generating and one for field definitions.

This first draft we put up here is not API-stable quite yet. And it's mostly focused on parsing existing messages. The next version should have a lot more support for mutating messages. 

Are you looking to generate messages from scratch? 

One issue we've found is that most of the subject matter experts seem to only refer to HL7 data by ordinal position, like PID-3.1. And HL7 messages in practice seem to violate the standards with wild abandon....

**Comment by DarkMarmot** [comment]
> I am looking to generate a message from scratch. A hint would be really great. I am new to Elixir so, I could have been staring at the solution without recognizing it.

Hi, I'm sorry for the very late reply -- this library was only originally designed for handling existing HL7 messages, and was getting changed a lot.

It's now settling down... though maybe/probably too late for your use case :(

If you use `mix docs` it will now generate a pretty good set of docs. It should be on hex.pm soon as well. 

You can now create messages using `HL7.Message.new` in tandem with `HL7.Header.new`.

`msg = HL7.Message.new(HL7.Header.new("ADT", "A01", "X1234"))`

This makes a message with only a header segment though.
You can then add segments to the Message struct and `to_string(msg)` when done.

To see what segments look like, try this in iex:

`HL7.Message.new(HL7.Examples.wikipedia_sample_hl7())`

Basically, each segment is a list of field data (strings or lists) -- and the first element is the segment type, e.g. "MSH" or "PID" o r "PV1".

If you generate the docs, most of what you might need is in HL7.Message and HL7.Query (if you want advanced stuff ever).

Good luck and my apologies for taking so long :(

-scott
